content/en/blog/_posts/2025-01-14-devices-failure-handling/index.md

```diff
@@ -0,0 +1,409 @@
+date: 2025-01-14
+slug: navigating-failures-in-pods-with-devices
+author: >
+  Sergey Kanzhelev (Google)
+  Mrunal Patel (RedHat)
+---
+
+Kubernetes is the de facto standard for container orchestration, but when it
+comes to handling specialized hardware like GPUs and other accelerators, things
+get a bit complicated. This blog post dives into the challenges of managing
+failure modes when operating pods with devices in Kubernetes, based on insights
+from [Sergey Kanzhelev and Mrunal Patel's talk at KubeCon NA
+2024](https://sched.co/1i7pT). You can follow the links to
+[slides](https://static.sched.com/hosted_files/kccncna2024/b9/KubeCon%20NA%202024_%20Navigating%20Failures%20in%20Pods%20With%20Devices_%20Challenges%20and%20Solutions.pptx.pdf?_gl=1*191m4j5*_gcl_au*MTU1MDM0MTM1My4xNzMwOTE4ODY5LjIxNDI4Nzk1NDIuMTczMTY0ODgyMC4xNzMxNjQ4ODIy*FPAU*MTU1MDM0MTM1My4xNzMwOTE4ODY5)
+and
+[recording](https://www.youtube.com/watch?v=-YCnOYTtVO8&list=PLj6h78yzYM2Pw4mRw4S-1p_xLARMqPkA7&index=150).
+
+## The AI/ML boom and its impact on Kubernetes
+
+The rise of AI/ML workloads has brought new challenges to Kubernetes. These
+workloads often rely heavily on specialized hardware, and any device failure can
+significantly impact performance and lead to frustrating interruptions. As
+highlighted in the 2024 [Llama
+paper](https://ai.meta.com/research/publications/the-llama-3-herd-of-models/),
+hardware issues, particularly GPU failures, are a major cause of disruption in
+AI/ML training. You can also learn how much effort NVIDIA spend on handing
+devices failures and maintenance in the KubeCon talk by [Ryan Hallisey and Piotr
+Prokop All-Your-GPUs-Are-Belong-to-Us: An Inside Look at NVIDIA's Self-Healing
+GeForce NOW
+Infrastructure](https://kccncna2024.sched.com/event/1i7kJ/all-your-gpus-are-belong-to-us-an-inside-look-at-nvidias-self-healing-geforce-now-infrastructure-ryan-hallisey-piotr-prokop-pl-nvidia)
+([recording](https://www.youtube.com/watch?v=iLnHtKwmu2I)) as they see 19
+remediation requests per 1000 nodes a day!
+We also see data centers offering spot consumption models and overcommit on
+power, making device failures commonplace and a part of the business model.
+
+However Kubernetes’s view on resources is still very static. The resource is
+either there or not. And if it is there, the assumption is that it will stay
+there fully functional - Kubernetes lacks good support for handling full or partial
+hardware failures. These long existing assumptions combined with the overall complexity of a set up, leads
+to a variety of failure modes, which we discuss here.
+
+### Understanding AI/ML workloads
+
+Generally, all AI/ML workloads need specialized hardware, have challenging
+scheduling requirements, and are expensive when idle. AI/ML workloads typically
+fall into two categories - training and inference. Here is an oversimplified
+view on those categories’ characteristics, which are different traditional,
+workloads like web services:
+
+Training
```

Comment:
Perhaps we should also mention that training usually involves distributed highly tied workloads (hint to the MPI Operator)